checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -243,3 +246,49 @@ def _create_new_foreach_resources(self, block_index_to_statement: FOR_EACH_BLOCK
             else:
                 self._create_new_resources_foreach(statement, self.local_graph.vertices[block_idx])
         self._delete_main_resource(block_index_to_statement)
+
+    def _handle_for_loop_in_foreach(self, block_idx: int, sub_graph: l_graph.TerraformLocalGraph) -> None:
+        block = sub_graph.vertices[block_idx]
+        foreach_statement = evaluate_terraform(block.attributes.get(FOREACH_STRING))
+        if not foreach_statement:
+            return
+
+        foreach_statement = self.extract_from_list(foreach_statement)
+        if isinstance(foreach_statement, str) and FOR_LOOP in foreach_statement:
+            if foreach_statement.startswith('${') and foreach_statement.endswith('}'):
+                foreach_statement = foreach_statement[2:-1]
+
+            start_bracket_idx = foreach_statement[1:].find(LEFT_BRACKET)
+            end_bracket_idx = find_match_bracket_index(foreach_statement, start_bracket_idx + 1)
+            if not start_bracket_idx != -1 and end_bracket_idx != -1:
+                return
+
+            rendered_foreach_statement = foreach_statement[start_bracket_idx:end_bracket_idx + 1].replace('"', '\\"').replace("'", '"')
+            if foreach_statement.startswith('{'):
+                rendered_foreach_statement = json.loads(rendered_foreach_statement)
+                block.attributes[FOREACH_STRING] = self._handle_for_loop_in_dict(rendered_foreach_statement, foreach_statement, end_bracket_idx + 1)
+            elif foreach_statement.startswith(LEFT_BRACKET):
+                rendered_foreach_statement = evaluate_terraform(rendered_foreach_statement)
+                block.attributes[FOREACH_STRING] = self._handle_for_loop_in_list(rendered_foreach_statement, foreach_statement, end_bracket_idx + 1)
+
+    @staticmethod
+    def _handle_for_loop_in_dict(object_to_run_on: list[dict[str, Any]], statement: str, start_expression_idx: int) -> dict[str, Any]:
+        expression = statement[start_expression_idx + 3:-1]  # 3 for 'bla bla' : -> " : " == 3
```

Comment:
I see this expression twice + added a comment for it, this suggest that it's not a trivial functionality.
I know it's only one line but still, better to extract to a function that abstracts this behavior.

Reply:
I agree, seems a bit hard to understand